### ISSUE TITLE
[4.0.0-alpha.2]: MUI warning when using AutocompleteInput

### ISSUE THREAD

**Issue by wattroll** [marmelab/react-admin#7061]
**What you were expecting:**
Use AutocompleteInput without warnings.

**What happened instead:**

Warning in the console
```
MUI: The value provided to Autocomplete is invalid. None of the options match with `""`. You can use the `isOptionEqualToValue` prop to customize the equality test.
```

Note the `""`, probably that's the offending part.

**Steps to reproduce:**
Seems to appear in pretty much any `AutocompleteInput`. Can be seen at posters galore demo, e.g. by going to Orders resource and adding a Customer filter.

---

I bisected it down to [this commit](https://github.com/marmelab/react-admin/pull/6971/commits/7de027dce7c3dff1fa11ca1210dfc7f7083b8858):

```
7de027dce7c3dff1fa11ca1210dfc7f7083b8858 is the first bad commit
commit 7de027dce7c3dff1fa11ca1210dfc7f7083b8858
Author: Travis CI <travis@travis-ci.org>
Date:   Wed Dec 15 14:44:42 2021 +0100

    Merge both autocomplete
```

I haven't managed to spot the cause of a problem by a quick look-through. @djhi, would you have time to have a look at this? :pray:

**Comment by WiXSL** [comment]
I couldn't reproduce the error you describe in the demo.
Could you provide a code sandbox or a repo to reproduce it?

**Comment by wattroll** [comment]
Thanks for having a look at it, and sorry for an incomplete bug report.

I tried, but haven't yet managed to setup a codesandbox against `next`. None of the vercel deployments that I found use AutocompletInput. I'll get back to it later.

In the meanwhile I made sure that I can reproduce locally in a demo with a fresh checkout:

1. `git clone 'https://github.com/marmelab/react-admin.git'`
2. `cd react-admin/`
2. `git checkout next`
3. `make build`
4. `make run-demo`
5. Open http://localhost:3000/#/commands
6. Click "Add filter" > "Customer"

![2022-01-10-140216_1920x1080_scrot](https://user-images.githubusercontent.com/79251424/148763879-ebf154bc-b7b0-4257-846f-ba442becc550.png)
